src/amo/containers/App.js

```diff
@@ -42,11 +47,30 @@ export class AppBase extends React.Component {
     InfoDialogComponent: InfoDialog,
     MastHeadComponent: MastHead,
     _addChangeListeners: addChangeListeners,
+    _navigator: (typeof navigator !== 'undefined' ? navigator : null),
     mozAddonManager: config.get('server') ? {} : navigator.mozAddonManager,
+    userAgent: null,
   }
 
   componentDidMount() {
-    const { _addChangeListeners, handleGlobalEvent, mozAddonManager } = this.props;
+    const {
+      _addChangeListeners,
+      _navigator,
+      handleGlobalEvent,
+      mozAddonManager,
+      setUserAgent,
+      userAgent,
+    } = this.props;
+
+    // If userAgent isn't set in state it could be that we couldn't get one
+    // from the request headers on our first (server) request. If that's the
+    // case we try to load them from navigator.
+    if (!userAgent && _navigator && _navigator.userAgent) {
+      log.info(
+        'userAgent not in state on App load; using navigator.userAgent.');
+      setUserAgent(_navigator.userAgent);
+    }
+
     // Use addonManager.addChangeListener to setup and filter events.
     _addChangeListeners(handleGlobalEvent, mozAddonManager);
```

Comment:
Hmm, I would feel better if this were called before `setUserAgent` because dispatching a redux action will cause re-rendering.